psutil/arch/freebsd/specific.c

```diff
@@ -1010,3 +1011,38 @@ psutil_sensors_battery(PyObject *self, PyObject *args) {
         PyErr_SetFromErrno(PyExc_OSError);
     return NULL;
 }
+
+
+/*
+ * Return temperature information.
+ */
+PyObject *
+psutil_sensors_temperatures(PyObject *self, PyObject *args) {
+    int current;
+    int tjmax;
+    int core;
+    char sensor[26];
+    size_t size = sizeof(current);
+
+    if (! PyArg_ParseTuple(args, "i", &core))
+        return NULL;
+    sprintf(sensor, "dev.cpu.%d.temperature", core);
+    if (sysctlbyname(sensor, &current, &size, NULL, 0))
+        goto error;
+    current = SYSCTLTEMP(current);
+
+    sprintf(sensor, "dev.cpu.%d.coretemp.tjmax", core);
+    if (sysctlbyname(sensor, &tjmax, &size, NULL, 0))
+        goto error;
```

Comment:
Considering we can have a valid "current" temp I think it's better to ignore this error and set "max" temp to `0`. Then from Python you can check if it's `0` and turn it into `None`.